lib/pbench/server/database/database.py

```diff
@@ -26,9 +26,11 @@ def get_engine_uri(config, logger):
 
     @staticmethod
     def init_db(server_config, logger):
-        # Attach the logger to the base class for models to find
+        # Attach the logger and server config object to the base class for
+        # database models to find
         if logger and not hasattr(Database.Base, "logger"):
             Database.Base.logger = logger
+            Database.Base.config = server_config
```

Comment:
Our whole logging infrastructure needs some work; we generally create a new logger with non-default options, and that's impossible to recreate without access to the config object, which we can really only get at the CLI level (e.g., it might be `_PBENCH_SERVER_CONFIG` or `-C` ... the former we could access anywhere, but not the latter). I suppose we should be modifying the default logger so that we can just `logging.getlogger()` in any arbitrary python module (which is how it's supposed to work).

Right now, having the base classes wired is expedient.

And, yeah, I probably shouldn't have nested the config setup under the logger test, although I'm not sure that the check isn't silly in the first place. Sigh.